File: noteworthy/lib/models/note.dart
class Note {
  late String id;
  late String title;
  late String description;
  late String userId;

  Note({
    required this.id,
    required this.title,
    required this.description,
    required this.userId,
  });

  factory Note.fromJson(Map<String, dynamic> json) {
    return Note(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      userId: json['userId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'content': description,
      'userId': userId,
    };
  }
}

File: noteworthy/backend/models/note.js
const mongoose = require('mongoose');
const User = require('./user');

const noteSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User', // Make sure the User model is defined and imported correctly
    required: true
  },
  title: {
    type: String,
    required: true // Uncomment if title is required
  },
  description: {
    type: String,
    required: true // Uncomment if content is required
  }
});

const Note = mongoose.model('Note', noteSchema);

module.exports = Note;
